R/doAzureParallel.R

```diff
@@ -253,174 +253,183 @@ setHttpTraffic <- function(value = FALSE) {
   assign("enableCloudCombine", enableCloudCombine, envir = .doAzureBatchGlobals)
   assign("cloudCombine", cloudCombine, envir = .doAzureBatchGlobals)
 
-  retryCounter <- 0
-  maxRetryCount <- 5
-  startupFolderName <- "startup"
-  containerResponse <- NULL
-  jobquotaReachedResponse <- NULL
-  while (retryCounter < maxRetryCount) {
-    sprintf("job id is: %s", id)
-    # try to submit the job. We may run into naming conflicts. If so, try again
-    tryCatch({
-      retryCounter <- retryCounter + 1
-
-      response <- rAzureBatch::createContainer(id, content = "text")
-      if (grepl("ContainerAlreadyExists", response)) {
-        if (!is.null(obj$options$azure$job)) {
-          containerResponse <- grepl("ContainerAlreadyExists", response)
-          break
-
-        }
+  resourceFiles <- list()
+  if (!is.null(obj$options$azure$resourceFiles)) {
+    resourceFiles <- obj$options$azure$resourceFiles
+  }
 
-        stop("Container already exists. Multiple jobs may possibly be running.")
-      }
+  if (!is.null(obj$options$azure$resourcefiles)) {
+    resourceFiles <- obj$options$azure$resourcefiles
+  }
 
-      rAzureBatch::uploadBlob(id,
-                              system.file(startupFolderName, "worker.R", package = "doAzureParallel"))
-      rAzureBatch::uploadBlob(id,
-                              system.file(startupFolderName, "merger.R", package = "doAzureParallel"))
-      rAzureBatch::uploadBlob(
-        id,
-        system.file(startupFolderName, "install_github.R", package = "doAzureParallel")
-      )
-      rAzureBatch::uploadBlob(
-        id,
-        system.file(startupFolderName, "install_cran.R", package = "doAzureParallel")
-      )
-      rAzureBatch::uploadBlob(
-        id,
-        system.file(startupFolderName, "install_bioconductor.R", package = "doAzureParallel")
-      )
+  enableCloudCombineKeyValuePair <-
+    list(name = "enableCloudCombine", value = as.character(enableCloudCombine))
 
-      # Setting up common job environment for all tasks
-      jobFileName <- paste0(id, ".rds")
-      saveRDS(.doAzureBatchGlobals, file = jobFileName)
+  chunkSize <- 1
 
-      rAzureBatch::uploadBlob(id, paste0(getwd(), "/", jobFileName))
+  if (!is.null(obj$options$azure$chunkSize)) {
+    chunkSize <- obj$options$azure$chunkSize
+  }
 
-      file.remove(jobFileName)
+  if (!is.null(obj$options$azure$chunksize)) {
+    chunkSize <- obj$options$azure$chunksize
+  }
 
-      resourceFiles <- list()
-      if (!is.null(obj$options$azure$resourceFiles)) {
-        resourceFiles <- obj$options$azure$resourceFiles
-      }
+  if (exists("chunkSize", envir = .doAzureBatchGlobals)) {
```

Comment:
If user specified chunkSize in foreach options, it will be overwritten by the chunkSize settings in the .doAzureBatchBlogals environment? Is this expected? Or we should reverse the sequence to give user specified options priority?

Reply:
Good point, I think if the user puts chunksize in the foreach, then it should overwrite it.